graphql/client.go

```diff
@@ -40,19 +42,68 @@ type Client interface {
 	) error
 }
 
+type WebSocketClient interface {
+	// StartWebSocket must open a webSocket connection and subscribe to an endpoint
+	// of the client's GraphQL API.
+	//
+	// errChan is a channel on which are sent the errors of webSocket
+	// communication.
+	//
+	// err is any error that occurs when setting up the webSocket connection.
+	StartWebSocket(ctx context.Context) (errChan chan error, err error)
+
+	// CloseWebSocket must close the webSocket connection. If no connection was
+	// started, CloseWebSocket is a no-op
+	CloseWebSocket()
+
+	// Subscribe must subscribe to an endpoint of the client's GraphQL API.
+	//
+	// req contains the data to be sent to the GraphQL server. Will be marshalled
+	// into JSON bytes.
+	//
+	// interfaceChan is a channel used to send the data that arrives via the
+	// webSocket connection.
+	//
+	// forwardDataFunc is the function that will cast the received interface into
+	// the valid type for the subscription's response.
+	//
+	// returns a subscriptionID if successful, an error otherwise
+	Subscribe(
+		req *Request,
+		interfaceChan interface{},
+		forwardDataFunc ForwardDataFunction,
+	) (string, error)
+
+	// Unsubscribe must unsubscribe from an endpoint of the client's GraphQL API.
+	Unsubscribe(subscriptionID string) error
+}
+
 type client struct {
 	httpClient Doer
 	endpoint   string
 	method     string
 }
 
+type webSocketClient struct {
```

Comment:
nit: probably clearer to move this + methods below to `websocket.go` (you can keep the interfaces here or not, either way)

Reply:
done, I kept the interface here next to the `Client`